src/pages/billing/NewBillingProjectWizard/GCPBillingProjectWizard/AddTerraAsBillingAccountUserStep.test.ts

```diff
@@ -0,0 +1,138 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { h } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+import * as Preferences from 'src/libs/prefs'
+import {
+  AddTerraAsBillingAccountUserStep,
+  AddTerraAsBillingAccountUserStepProps
+} from 'src/pages/billing/NewBillingProjectWizard/GCPBillingProjectWizard/AddTerraAsBillingAccountUserStep'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+jest.mock('src/libs/ajax')
+jest.spyOn(Preferences, 'getLocalPref')
+
+type AjaxContract = ReturnType<typeof Ajax>
+
+function expectNotNull<T>(value: T | null): T {
+  expect(value).not.toBeNull()
+  return value as T
+}
+
+
+const textMatcher = text => screen.queryByText((_, node) => {
+  const hasText = node => node.textContent === text
+  const nodeHasText = hasText(node)
+  const childrenDontHaveText = Array.from(node?.children || []).every(
+    child => !hasText(child)
+  )
+  return nodeHasText && childrenDontHaveText
+})
+
+
+const getNoAccessButton = () => expectNotNull(screen.queryByLabelText("I don't have access to do this"))
+const getUserAddedButton = () => expectNotNull(screen.queryByLabelText('I have added terra-billing as a billing account user (requires reauthentication)'))
+
+const createGCPProject = jest.fn(() => Promise.resolve())
+const captureEvent = jest.fn()
+describe('AddTerraAsBillingAccountUserStep', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+
+    asMockedFn(Ajax).mockImplementation(() => ({
+      Billing: { createGCPProject } as Partial<AjaxContract['Billing']>,
+      Metrics: { captureEvent } as Partial<AjaxContract['Metrics']>
+    } as Partial<AjaxContract> as AjaxContract))
+  })
+
+
+  describe('The initial state when the step is active', () => {
+    beforeEach(() => {
+      render(h(AddTerraAsBillingAccountUserStep, {
+        isActive: true,
+        setAccessToAddBillingAccountUser: jest.fn,
+        isFinished: false,
+      }))
+    })
+
+    it('has both radio button options shown, with neither selected ', () => {
+      const noAccessButton = screen.queryByLabelText("I don't have access to do this")
+      expect(noAccessButton).toBeDefined()
+      expect(noAccessButton).not.toBeChecked()
+      expect(noAccessButton).not.toBeDisabled()
+      const terraUserAddedButton = screen.queryByLabelText('I have added terra-billing as a billing account user (requires reauthentication)')
+      expect(terraUserAddedButton).toBeDefined()
+      expect(terraUserAddedButton).not.toBeChecked()
+      expect(terraUserAddedButton).not.toBeDisabled()
+    })
+
+    it('shows the correct instructions to the user', () => {
+      const instructions = textMatcher('Add terra-billing@terra.bio as a Billing Account User to your billing account.')
```

Comment:
Testing the full text that is displayed feels like a bit too much to me (as well as testing that some controls are radio buttons). @msilva-broad do you think it is possible for unit tests to get _too_ specific/detailed?